02-peer-protocol.md

```diff
@@ -787,9 +787,21 @@ is destined, is described in [BOLT #4](04-onion-routing.md).
 #### Requirements
 
 A sending node:
-  - MUST NOT offer `amount_msat` it cannot pay for in the
-remote commitment transaction at the current `feerate_per_kw` (see "Updating
-Fees") while maintaining its channel reserve.
+  - if it is _responsible_ for paying the Bitcoin fee:
+    - MUST NOT offer `amount_msat` if, after adding that HTLC to its commitment
+    transaction, it cannot pay the fee for either the local or remote commitment
+    transaction at the current `feerate_per_kw` while maintaining its channel
+    reserve (see [Updating Fees](#updating-fees-update_fee)).
+    - SHOULD NOT offer `amount_msat` if, after adding that HTLC to its commitment
+    transaction, its remaining balance doesn't allow it to pay the fee for a
+    future additional non-dust HTLC at a higher feerate while maintaining its
+    channel reserve ("fee spike buffer"). A buffer of `2*feerate_per_kw` is
+    recommended to ensure predictability.
```

Comment:
I suppose the wording is not clear enough, given both c-lightning and eclair now implemented this, but in incompatible ways due to different interpretations.

- The wording here is basically "ctx plus 1 more HTLC at twice the current feerate".
- [In eclair](https://github.com/ACINQ/eclair/blame/eae113f0988069044dc268d8a7f16728563baa5f/eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala#L192-L198), the implementation checks "ctx plus (1 more HTLC at twice the current feerate)":
there is an "additional reserve", which is the on-chain fee cost of 2 more HTLC outputs at the current feerate.
- [In c-lightning](https://github.com/ElementsProject/lightning/blob/af7e87930878117a0633bb7a8cd4488b8e9c5b6d/channeld/full_channel.c#L430-L440) however, the check is "(ctx plus 1 more HTLC) at twice the current feerate":
there is an "additional reserve", set by calculating how much fees the funder would need to pay for the whole commitment transaction (with an extra dummy HTLC).

(to be clear, in both cases, the "ctx" already includes the HTLC being offered, and the "1 more HTLC" is an additional one)

Reply:
Interesting, that means we'll have to discuss this further on IRC tonight to get more bandwidth than discussing it on Github ;)
Feel free to suggest a better wording in the meantime though.

Reply:
I don't know what the wording should be because I don't know what the intended meaning is. :)

Reply:
I think c-lighting got it right. An increase of the feerate must be applied to the whole tx, not only an htlc, so my implementation in eclair is a bit weird. I tried rephrasing in https://github.com/lightningnetwork/lightning-rfc/pull/740/commits/afb7313e0f81bfee32aaa9b71fbf6c512407e8d1, I hope it's clearer. Feel free to suggest something if still unclear (I'm still not 100% happy with what I came up with).

Reply:
What about now? Do you have a suggestion on how to make it more explicit @SomberNight?